File: menumodule/src/main/java/zjm/cst/dhu/menumodule/FileTypeContract.java
package zjm.cst.dhu.menumodule;

import java.util.List;

import zjm.cst.dhu.basemodule.model.EncryptRelation;
import zjm.cst.dhu.basemodule.model.EncryptType;
import zjm.cst.dhu.basemodule.model.File;
import zjm.cst.dhu.basemodule.presenter.BasePresenter;
import zjm.cst.dhu.basemodule.view.BaseView;

/**
 * Created by zjm on 2017/3/3.
 */

public interface FileTypeContract {
    interface View extends BaseView {
        void updateEncryptType(List<EncryptType> list);

        void getFileTypeNetworkError();

        void typeDetailClick(EncryptType encryptType);

        void setFile(File file);

        void setDesKey(EncryptRelation encryptRelation);

        void encryptBaseTypeNetworkError();

        void encryptBaseTypeEncryptSuccess();

        void decryptFileExistError();

        void decryptBaseTypeDecryptSuccess();

        void decryptBaseTypeDecryptError();

        void decryptBaseTypeDecryptFailed();

        void confirmDesKey(EncryptRelation encryptRelation);
    }

    interface Presenter extends BasePresenter<View> {
        void getEncryptType();

        void encryptFile(EncryptRelation encryptRelation);

        void encryptBaseType(EncryptRelation encryptRelation, String desKey, String desLayer);

        void decryptFile(EncryptRelation encryptRelation, File file);

        void decryptBaseType(File file, String desKey);
    }
}

File: menumodule/src/main/java/zjm/cst/dhu/menumodule/FileTypeContract.kt
package zjm.cst.dhu.menumodule

import zjm.cst.dhu.basemodule.model.EncryptRelation
import zjm.cst.dhu.basemodule.model.EncryptType
import zjm.cst.dhu.basemodule.model.File
import zjm.cst.dhu.basemodule.presenter.BasePresenter
import zjm.cst.dhu.basemodule.view.BaseView

/**
 * Created by zjm on 2017/3/3.
 */

interface FileTypeContract {
    interface View : BaseView {
        fun updateEncryptType(list: List<EncryptType>)

        fun getFileTypeNetworkError()

        fun setEncryptTypeRateNetworkError()

        fun typeDetailClick(encryptType: EncryptType)

        fun setFile(file: File)

        fun setDesKey(encryptRelation: EncryptRelation)

        fun encryptBaseTypeNetworkError()

        fun encryptBaseTypeEncryptSuccess()

        fun decryptFileExistError()

        fun decryptBaseTypeDecryptSuccess()

        fun decryptBaseTypeDecryptError()

        fun decryptBaseTypeDecryptFailed()

        fun confirmDesKey(encryptRelation: EncryptRelation)
    }

    interface Presenter : BasePresenter<View> {
        fun getEncryptType()

        fun getEncryptTypeByOwner(id: Int)

        fun setEncryptTypeRate(userId: Int, typeId: Int, rate: Float)

        fun encryptFile(encryptRelation: EncryptRelation)

        fun encryptBaseType(encryptRelation: EncryptRelation, desKey: String, desLayer: String)

        fun decryptFile(encryptRelation: EncryptRelation, file: File?)

        fun decryptBaseType(file: File?, desKey: String)
    }
}
